samples/drivers/spi_flash/src/main.c

```diff
@@ -56,6 +57,19 @@ void single_sector_test(const struct device *flash_dev)
 		printf("Flash erase succeeded!\n");
 	}
 
+	/* Check that beginning of sector is actually erased. */
+	memset(buf, 0, sizeof(buf));
+	rc = flash_read(flash_dev, SPI_FLASH_TEST_REGION_OFFSET, buf, len);
```

Comment:
If you really want to check whether sector has been erased, you need to loop through entire sector. The fact that first few bytes are erased does not prove the sector is.

Reply:
The intention was not to check that the sector was erased. Only the four bytes that are later written.

Reply:
Fair enough.